File: 백준/Silver/1463. 1로 만들기/1로 만들기.java
import java.util.*;

public class Main {

    public static int dp(int n) {
        int[] arr = new int[(n <= 3?4:n+1)];

        arr[1] = 0;
        arr[2] = 1;
        arr[3] = 1;

        for (int i = 4; i <= n; i++) {

            int tmp1 = Integer.MAX_VALUE;
            int tmp2 = Integer.MAX_VALUE;
            int tmp3 = Integer.MAX_VALUE;

            if (i % 3 == 0){
                tmp1 = 1 + arr[i / 3];
            }

            if (i % 2 == 0) {
                tmp2 = 1 + arr[i / 2];
            }

            tmp3 = 1 + arr[i - 1];

            arr[i] = Math.min(Math.min(tmp1, tmp2), tmp3);

        }

        return arr[n];
    }

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        int x = sc.nextInt();
        System.out.println(dp(x));

    }
}

File: 백준/Silver/1463. 1로 만들기/1로 만들기.py
N = int(input())
A = [0] * (N + 1)
A[1] = 0

for i in range(2, N + 1):
    A[i] = 1 + A[i - 1]
    if i % 3 == 0:
        A[i] = min(A[i], 1 + A[i // 3])
    if i % 2 == 0:
        A[i] = min(A[i], 1 + A[i // 2])
print(A[N])